pkg/controllers/provisioningv2/rke2/provisioningcluster/controller.go

```diff
@@ -168,33 +210,87 @@ func (h *handler) OnRancherClusterChange(obj *rancherv1.Cluster, status rancherv
 		return nil, status, nil
 	}
 
-	status, err := h.updateClusterProvisioningStatus(obj, status)
-	if err != nil && !apierror.IsNotFound(err) {
+	rkeCP, err := h.getRKEControlPlaneForCluster(obj)
+	if err != nil {
 		return nil, status, err
 	}
 
+	// If the rkecontrolplane is not nil, we can check it to determine action items.
+	if rkeCP != nil {
+		// If EtcdSnapshotRestore is not nil, we need to check to see if we need to update the cluster object it.
+		if obj.Spec.RKEConfig.ETCDSnapshotRestore != nil &&
+			obj.Spec.RKEConfig.ETCDSnapshotRestore.Name != "" &&
+			(obj.Spec.RKEConfig.ETCDSnapshotRestore.RestoreRKEConfig != "" &&
+				obj.Spec.RKEConfig.ETCDSnapshotRestore.RestoreRKEConfig != restoreRKEConfigNone) {
+			logrus.Debugf("rkecluster %s/%s: Reconciling rkeconfig against specified etcd restore snapshot metadata", obj.Namespace, obj.Name)
+			if !equality.Semantic.DeepEqual(rkeCP.Status.ETCDSnapshotRestore, obj.Spec.RKEConfig.ETCDSnapshotRestore) {
+				clusterSpec, err := h.findSnapshotClusterSpec(obj.Namespace, fmt.Sprintf("%s-%s", obj.Name, obj.Spec.RKEConfig.ETCDSnapshotRestore.Name))
+				if err != nil {
+					return nil, status, err
+				}
+				switch obj.Spec.RKEConfig.ETCDSnapshotRestore.RestoreRKEConfig {
```

Comment:
No, because that is not flexible and cannot be easily reconciled in the future. If a feature request comes in for a specific functionality of restoring cluster data, we will need to capture that case and should already have the framework in place to handle it.